### PR TITLE
Auto detect backend2

### PR DESCRIPTION
**Pull request by bendichter**

### CODE REVIEW

**Comment by bendichter** [comment]
@CodyCBakerPhD how would you feel about this? It's a bit friendlier in that if you do specify both and they match you get a warning instead of an error

File: src/neuroconv/basedatainterface.py [hunk 1]
```diff
@@ -153,16 +153,24 @@ def run_conversion(
...
-            raise ValueError("Please specify only one of `backend` or `backend_configuration`.")
+            if backend == backend_configuration.backend:
+                warnings.warn(
+                    f"Both `backend` and `backend_configuration` were specified as type '{backend}'. "
+                    "To suppress this warning, specify only `backend_configuration`."
+                )
+            else:
+                raise ValueError(
+                    f"Both `backend` and `backend_configuration` were specified and are conflicting."
+                    f"backend: {backend}, backend_configuration.backend: {backend_configuration.backend}."
```
Comment (CodyCBakerPhD) [inline src/neuroconv/basedatainterface.py:164]:
```suggestion
                    f"{backend=}, {backend_configuration.backend=}."
```

**Comment by CodyCBakerPhD** [comment]
Fine by me - small suggestion for printing syntax that was added in Python 3.8

File: src/neuroconv/nwbconverter.py [hunk 2]
```diff
@@ -194,16 +195,24 @@ def run_conversion(
...
-            raise ValueError("Please specify only one of `backend` or `backend_configuration`.")
+            if backend == backend_configuration.backend:
+                warnings.warn(
+                    f"Both `backend` and `backend_configuration` were specified as type '{backend}'. "
+                    "To suppress this warning, specify only `backend_configuration`."
+                )
+            else:
+                raise ValueError(
+                    f"Both `backend` and `backend_configuration` were specified and are conflicting."
+                    f"backend: {backend}, backend_configuration.backend: {backend_configuration.backend}."
```
Comment (bendichter) [inline src/neuroconv/nwbconverter.py:206]:
```suggestion
                    f"{backend=}, {backend_configuration.backend=}."
```

**Pull request merged**